python/ray/util/multiprocessing/pool.py

```diff
@@ -334,29 +355,52 @@ def __init__(self, pool, func, iterable, chunksize=None):
         self._pool = pool
         self._func = func
         self._next_chunk_index = 0
+        self._finished_iterating = False
         # List of bools indicating if the given chunk is ready or not for all
         # submitted chunks. Ordering mirrors that in the in the ResultThread.
         self._submitted_chunks = []
         self._ready_objects = collections.deque()
-        if not hasattr(iterable, "__len__"):
+        try:
+            self._iterator = iter(iterable)
+        except TypeError:
+            # for compatibility with prior releases, encapsulate non-iterable in a list
             iterable = [iterable]
-        self._iterator = iter(iterable)
-        self._chunksize = chunksize or pool._calculate_chunksize(iterable)
-        self._total_chunks = div_round_up(len(iterable), chunksize)
-        self._result_thread = ResultThread([], total_object_refs=self._total_chunks)
+            self._iterator = iter(iterable)
+        if self._iterator == iterable:
+            # we were passsed an iterator, so do not know the number of samples
+            self._chunksize = chunksize or 1
+            result_list_size = 0  # len(self._pool._actor_pool)
```

Comment:
I'm not sure what the comment means– if it's leftover code, can we remove it?

Reply:
Yes, you're right!
```suggestion
            result_list_size = 0
```